File: HandSock-Server/src/main/java/com/server/handsock/sockets/handler/ClientHandler.java
package com.server.handsock.sockets.handler;

import com.corundumstudio.socketio.AckRequest;
import com.server.handsock.console.HandleResults;
import com.server.handsock.services.ClientService;
import com.corundumstudio.socketio.SocketIOClient;
import com.server.handsock.clients.service.ClientUserService;

import java.util.HashMap;
import java.util.Map;

public class ClientHandler {

    private final Map<String, Object> service;
    private final Map<String, Object> clientServiceList;
    private final HandleResults handleResults = new HandleResults();

    public ClientHandler(Map<String, Object> service, Map<String, Object> clientServiceList) {
        this.service = service;
        this.clientServiceList = clientServiceList;
    }

    public void handleClientInit(SocketIOClient client, Map<String, Object> data, AckRequest ackRequest) {
        ClientService clientService = new ClientService(service, clientServiceList);
        try {
            if (clientService.validClientToken(client)) {
                Map<String, Object> result = new HashMap<>();
                ClientUserService clientUserService = (ClientUserService) clientServiceList.get("clientUserService");
                result.put("userinfo", clientUserService.queryUserInfo(clientService.getRemoteUID(client)));
                ackRequest.sendAckData(handleResults.handleResultByCode(200, result, "获取成功"));
            } else ackRequest.sendAckData(handleResults.handleResultByCode(403, null, "请重新登录"));
        } catch (Exception e) {
            clientService.handleException(e, ackRequest);
        }
    }

}

File: HandSock-Server/src/main/java/com/server/handsock/sockets/handler/ClientHandler.kt
package com.server.handsock.sockets.handler

import com.corundumstudio.socketio.AckRequest
import com.corundumstudio.socketio.SocketIOClient
import com.server.handsock.clients.service.ClientUserService
import com.server.handsock.services.AuthService
import com.server.handsock.services.ClientService
import com.server.handsock.utils.HandUtils
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ClientHandler @Autowired constructor(
    private val authService: AuthService,
    private val clientService: ClientService,
    private val clientUserService: ClientUserService
) {
    fun handleClientInit(client: SocketIOClient, ackRequest: AckRequest) {
        authService.validClientStatusBySocket(client) {
            val result: MutableMap<String, Any> = HashMap()
            result["userinfo"] = clientUserService.queryUserInfo(clientService.getRemoteUID(client))
            ackRequest.sendAckData(HandUtils.handleResultByCode(200, result, "获取成功"))
        }
    }
}
